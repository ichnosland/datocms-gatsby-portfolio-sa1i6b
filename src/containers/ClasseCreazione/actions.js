/*
 *
 * ClasseCreazione actions
 *
 */

import {
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
  CLASSE_CREAZIONE_GEO_SET,
  CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  CLASSE_CREAZIONE_GEO_COMUNI_SET,
  CLASSE_CREAZIONE_DATA_SET,
  CLASSE_CREAZIONE_DATA_POST,
  CLASSE_CREAZIONE_RESET,
  CLASSE_CREAZIONE_SPINNER_SET,
  CLASSE_CREAZIONE_FEEDBACK_SET,
  CLASSE_CREAZIONE_FEEDBACK_RESET,
  CLASSE_CREAZIONE_GEO_SCUOLE_SET,
} from './constants';


export const classeCreazioneScuoleAttiveFetch = () => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
});

export const classeCreazioneScuoleAttiveSet = (payload) => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  payload,
});

export const classeCreazioneGeoProvinciaFetch = () => ({
  type: CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
});

export const classeCreazioneGeoSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SET,
  payload,
});

export const classeCreazioneGeoComuneFetch = (sigla) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  sigla,
});

export const classeCreazioneGeoComuneSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_SET,
  payload,
});

export const classeCreazioneGeoScuoleSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SCUOLE_SET,
  payload,
});

export const classeCreazioneDataSet = (payload) => ({
  type: CLASSE_CREAZIONE_DATA_SET,
  payload,
});

export const classeCreazioneDataPost = (payload, history) => ({
  type: CLASSE_CREAZIONE_DATA_POST,
  payload,
  history,
});

export const classeCreazioneReset = () => ({
  type: CLASSE_CREAZIONE_RESET,
});

export const classeCreazioneSpinnerSet = (enable) => ({
  type: CLASSE_CREAZIONE_SPINNER_SET,
  enable,
});

export const classeCreazioneFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSE_CREAZIONE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classeCreazioneFeedbackReset = () => ({
  type: CLASSE_CREAZIONE_FEEDBACK_RESET,
});
