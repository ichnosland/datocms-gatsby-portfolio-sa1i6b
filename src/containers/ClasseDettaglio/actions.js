/*
 *
 * ClasseDettaglio actions
 *
 */

import {
  CLASSE_DETTAGLIO_ESPELLI,
  CLASSE_DETTAGLIO_RESET,
  CLASSE_DETTAGLIO_DATA_FETCH,
  CLASSE_DETTAGLIO_CORSO_FETCH,
  CLASSE_DETTAGLIO_CONTENUTO_SET,
  CLASSE_DETTAGLIO_DISPLAY_SET,
  CLASSE_DETTAGLIO_DISPLAY_SORT,
  CLASSE_DETTAGLIO_FEEDBACK_SET,
  CLASSE_DETTAGLIO_FEEDBACK_RESET,
  CLASSE_DETTAGLIO_SPINNER_SET,
} from './constants';

export const classeDettaglioEspelli = (payload) => ({
  type: CLASSE_DETTAGLIO_ESPELLI,
  payload,
});

export const classeDettaglioReset = () => ({
  type: CLASSE_DETTAGLIO_RESET,
});

export const classeDettaglioCorsoFetch = (payload) => ({
  type: CLASSE_DETTAGLIO_CORSO_FETCH,
  payload,
});

export const classeDettaglioDataFetch = (payload) => ({
  type: CLASSE_DETTAGLIO_DATA_FETCH,
  payload,
});

export const classeDettaglioContenutoSet = (payload) => ({
  type: CLASSE_DETTAGLIO_CONTENUTO_SET,
  payload,
});

export const classeDettaglioDisplaySort = (payloadData, sortingData) => ({
  type: CLASSE_DETTAGLIO_DISPLAY_SORT,
  payloadData,
  sortingData,
});

export const classeDettaglioDisplaySet = (payload) => ({
  type: CLASSE_DETTAGLIO_DISPLAY_SET,
  payload,
});

export const classeDettaglioFeedbackSet = (hasFeedback, tipologia, messaggio) => ({
  type: CLASSE_DETTAGLIO_FEEDBACK_SET,
  hasFeedback,
  tipologia,
  messaggio,
});

export const classeDettaglioFeedbackReset = () => ({
  type: CLASSE_DETTAGLIO_FEEDBACK_RESET,
});

export const classeDettaglioSpinnerSet = (enable) => ({
  type: CLASSE_DETTAGLIO_SPINNER_SET,
  enable,
});
