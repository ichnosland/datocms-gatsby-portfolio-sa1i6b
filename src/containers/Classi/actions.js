/*
 *
 * Classi actions
 *
 */

import {
  CLASSI_DATA_FETCH,
  CLASSI_DATA_SET,
  CLASSI_DATA_RESET,
  CLASSI_FEEDBACK_RESET,
  CLASSI_FEEDBACK_SET,
  CLASSI_SPINNER_SET,
} from './constants';

export const classiDataFetch = (idDisciplina) => ({
  type: CLASSI_DATA_FETCH,
  idDisciplina,
});

export const classiDataSet = (payload) => ({
  type: CLASSI_DATA_SET,
  payload,
});

export const classiDataReset = () => ({
  type: CLASSI_DATA_RESET,
});

export const classiFeedbackReset = () => ({
  type: CLASSI_FEEDBACK_RESET,
});

export const classiFeedbackSet = (enable = true, message = '', errorData = {}) => ({
  type: CLASSI_FEEDBACK_SET,
  enable,
  message,
  errorData,
});

export const classiSpinnerSet = (enable) => ({
  type: CLASSI_SPINNER_SET,
  enable,
});
