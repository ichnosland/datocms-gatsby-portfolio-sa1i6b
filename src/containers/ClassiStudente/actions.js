/*
 *
 * ClassiStudente actions
 *
 */

import {
  CLASSI_STUDENTE_FETCH,
  CLASSI_STUDENTE_SET,
  CLASSI_STUDENTE_RESET,
  CLASSI_STUDENTE_ISCRIVITI_POST,
  CLASSI_STUDENTE_SPINNER_SET,
  CLASSI_STUDENTE_FEEDBACK_SET,
  CLASSI_STUDENTE_FEEDBACK_RESET,
} from './constants';

export const classiStudenteFetch = (disciplinaId) => ({
  type: CLASSI_STUDENTE_FETCH,
  disciplinaId,
});

export const classiStudenteSet = (payload) => ({
  type: CLASSI_STUDENTE_SET,
  payload,
});

export const classiStudenteReset = () => ({
  type: CLASSI_STUDENTE_RESET,
});

export const classiStudenteIscrivitiPost = (codiceCorso, configuration) => ({
  type: CLASSI_STUDENTE_ISCRIVITI_POST,
  codiceCorso,
  configuration,
});

export const classiStudenteSpinnerSet = (enable) => ({
  type: CLASSI_STUDENTE_SPINNER_SET,
  enable,
});

export const classiStudenteFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSI_STUDENTE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classiStudenteFeedbackReset = () => ({
  type: CLASSI_STUDENTE_FEEDBACK_RESET,
});
