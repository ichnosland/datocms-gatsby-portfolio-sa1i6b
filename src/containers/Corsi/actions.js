/*
 *
 * Corsi actions
 *
 */

import {
  CORSI_RESET,
  CORSI_ERROR_SET,
  CORSI_ERROR_RESET,
  CORSI_SPINNER_SET,
  CORSI_LIST_INITIALIZE,
  CORSI_LIST_DOCENTE_FETCH,
  CORSI_LIST_STUDENTE_FETCH,
  CORSI_LIST_SET,
  CORSI_CORSO_DETAIL_DOCENTE_TRIGGER,
  CORSI_CORSO_DETAIL_DOCENTE_SET,
} from './constants';

export const corsiReset = () => ({
  type: CORSI_RESET,
});

export const corsiSpinnerSet = (enable) => ({
  type: CORSI_SPINNER_SET,
  enable,
});

export const corsiErrorSet = (enable = true, message = '', errorData = {}) => ({
  type: CORSI_ERROR_SET,
  enable,
  message,
  errorData,
});

export const corsiErrorReset = () => ({
  type: CORSI_ERROR_RESET,
});

export const corsiListInitialize = (configuration, isDocente, userId) => ({
  type: CORSI_LIST_INITIALIZE,
  configuration,
  isDocente,
  userId,
});

export const corsiListDocenteFetch = (idDisciplina) => ({
  type: CORSI_LIST_DOCENTE_FETCH,
  idDisciplina,
});

export const corsiListStudenteFetch = (idDisciplina) => ({
  type: CORSI_LIST_STUDENTE_FETCH,
  idDisciplina,
});

export const corsiListSet = (payload) => ({
  type: CORSI_LIST_SET,
  payload,
});

export const corsiCorsoSelectedDocenteTrigger = (idCorso, configuration, isDocente, userId) => ({
  type: CORSI_CORSO_DETAIL_DOCENTE_TRIGGER,
  idCorso,
  configuration,
  isDocente,
  userId,
});

export const corsiCorsoDetailDocenteSet = (payload) => ({
  type: CORSI_CORSO_DETAIL_DOCENTE_SET,
  payload,
});
