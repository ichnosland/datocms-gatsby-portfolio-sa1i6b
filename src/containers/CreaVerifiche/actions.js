/*
 *
 * CreaVerifiche actions
 *
 */

import {
  CREA_VERIFICHE_FETCH,
  CREA_VERIFICHE_FETCH_ERROR,
  CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH,
  CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH_ERROR,
  CREA_VERIFICHE_SET,
  CREA_VERIFICHE_SPINNER_SET,
  CREA_VERIFICHE_POST,
  CREA_VERIFICHE_POST_ERROR,
  CREA_VERIFICHE_VERIFICA_SET,
  CREA_VERIFICHE_VERIFICA_ESERCIZIO_ADD,
  CREA_VERIFICHE_VERIFICA_ESERCIZIO_REMOVE,
  CREA_VERIFICHE_VERIFICA_RIEPILOGO_VIEW,
} from './constants';

export const creaverificheVerificaPostError = (messaggio = 'Impossibile inviare i dati') => ({
  type: CREA_VERIFICHE_POST_ERROR,
  messaggio,
});

export const creaverificheVerificaPost = (payload) => ({
  type: CREA_VERIFICHE_POST,
  payload,
});

export const creaverificheVerificaEserciziFetch = (prerequisito) => ({
  type: CREA_VERIFICHE_FETCH,
  prerequisito,
});

export const creaverificheVerificaEserciziFetchError = (messaggio = 'Impossibile caricare l\'elenco degli esercizi disponibili') => ({
  type: CREA_VERIFICHE_FETCH_ERROR,
  messaggio,
});

export const creaverificheVerificaEsercizioPreviewFetch = (esercizioId) => ({
  type: CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH,
  esercizioId,
});

export const creaverificheVerificaEsercizioPreviewFetchError = (
  messaggio = 'Impossibile caricare l\'elenco degli step per questo esercizio'
) => ({
  type: CREA_VERIFICHE_PREVIEW_ESERCIZIO_FETCH_ERROR,
  messaggio,
});

export const creaverificheVerificaEserciziSet = (payload) => ({
  type: CREA_VERIFICHE_SET,
  payload,
});

export const creaverificheVerificaEserciziSpinnerSet = (enabled) => ({
  type: CREA_VERIFICHE_SPINNER_SET,
  enabled,
});

export const creaverificheVerificaSet = (payload) => ({
  type: CREA_VERIFICHE_VERIFICA_SET,
  payload,
});

export const creaverificheVerificaEsercizioAdd = (step) => ({
  type: CREA_VERIFICHE_VERIFICA_ESERCIZIO_ADD,
  step,
});

export const creaverificheVerificaEsercizioRemove = (step) => ({
  type: CREA_VERIFICHE_VERIFICA_ESERCIZIO_REMOVE,
  step,
});

export const creaverificheVerificaRiepilogoView = (enabled) => ({
  type: CREA_VERIFICHE_VERIFICA_RIEPILOGO_VIEW,
  enabled,
});
