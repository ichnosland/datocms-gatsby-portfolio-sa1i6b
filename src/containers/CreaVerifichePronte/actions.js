/*
 *
 * CreaVerifichePronte actions
 *
 */

import {
  CREA_VERIFICHE_PRONTE_FETCH,
  CREA_VERIFICHE_PRONTE_FETCH_ERROR,
  CREA_VERIFICHE_PRONTE_SET,
  CREA_VERIFICHE_PRONTE_SPINNER_SET,
  CREA_VERIFICHE_PRONTE_FETCH_ESERCIZI,
} from './constants';

export const creaverificheVerifichePronteFetch = (disciplinaId) => ({
  type: CREA_VERIFICHE_PRONTE_FETCH,
  disciplinaId,
});

export const creaverificheVerifichePronteFetchError = (messaggio = 'Impossibile caricare l\'elenco delle verifiche pronte') => ({
  type: CREA_VERIFICHE_PRONTE_FETCH_ERROR,
  messaggio,
});

export const creaverificheVerifichePronteSet = (payload) => ({
  type: CREA_VERIFICHE_PRONTE_SET,
  payload,
});

export const creaverificheVerifichePronteSpinnerSet = (enabled) => ({
  type: CREA_VERIFICHE_PRONTE_SPINNER_SET,
  enabled,
});

export const creaverificheVerifichePronteEserciziFetch = (payload) => ({
  type: CREA_VERIFICHE_PRONTE_FETCH_ESERCIZI,
  payload,
});
