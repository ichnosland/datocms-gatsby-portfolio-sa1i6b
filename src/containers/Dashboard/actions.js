import {
  DASHBOARD_UNITA_FILTRA,
  DASHBOARD_LIVELLI_SET,
  DASHBOARD_LIVELLI_FETCH,
  DASHBOARD_LIVELLI_FETCH_ERROR,
  DASHBOARD_SPINNER_SET,
  DASHBOARD_LIVELLI_RESET,
  DASHBOARD_MENUEGG_SET,
  DASHBOARD_SEARCHACTIVE_SET,
  DASHBOARD_LIVELLI_ISLOADING_SET,
} from './constants';

export function livelliFetch(configuration, isDocente = false, corsoId = undefined) {
  return {
    type: DASHBOARD_LIVELLI_FETCH,
    configuration,
    isDocente,
    corsoId,
  };
}

export function livelliSet(payload) {
  return {
    type: DASHBOARD_LIVELLI_SET,
    payload,
  };
}

export function livelliReset() {
  return {
    type: DASHBOARD_LIVELLI_RESET,
  };
}

export function livelliFetchError(message = 'Errore caricamento') {
  return {
    type: DASHBOARD_LIVELLI_FETCH_ERROR,
    message,
  };
}

export function livelliIsLoadingSet(value) {
  return {
    type: DASHBOARD_LIVELLI_ISLOADING_SET,
    value,
  };
}

export function spinnerSet(payload) {
  return {
    type: DASHBOARD_SPINNER_SET,
    payload,
  };
}

export function unitaFiltra(parolaChiave) {
  return {
    type: DASHBOARD_UNITA_FILTRA,
    filtro: parolaChiave,
  };
}

export const dashboardMenuEggSet = (value) => ({
  type: DASHBOARD_MENUEGG_SET,
  value,
});

export const dashboardSearchActiveSet = (value) => ({
  type: DASHBOARD_SEARCHACTIVE_SET,
  value,
});
