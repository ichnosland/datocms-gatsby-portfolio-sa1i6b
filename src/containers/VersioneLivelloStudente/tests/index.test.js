import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'configureStore';
import { compose } from 'redux';
import { connect } from 'react-redux';

import injectReducer from 'utils/injectReducer';

import reducerCurryer from 'common/applications/reducer';
import Spinner from 'components/Spinner';
import { AlertBanner } from 'components/AlertBanner';
import versioneReducer, {
  defaultVersioneFeedback,
  defaultVersioniLivello,
} from 'containers/Versioni/reducer';
import { versioniLivelloFetch, versioniVersioneEsecuzioneInitialize } from 'containers/Versioni/actions';
import userReducer, { defaultAppData } from 'containers/User/reducer';

import VersioneStudente, { VersioniStudenteView } from '../index';
import VersioniLivelloOverview from '../LivelloStudenteOverview';


const mockConfiguration = {
  product: 'alatin',
  hasPremium: true,
  disciplinaId: 777,
  switcherMyTest: true,
  homePage: '/homepage',
};

const mockProps = {
  spinner: false,
  match: {
    params: {
      id: '123',
    },
  },
  history: {
    push: () => { },
  },
  configuration: mockConfiguration,
  feedback: {
    hasFeedback: false,
    tipologia: '',
    messaggio: '',
  },
  versioniLivello: {
    spinnerSmall: {},
    isLoaded: true,
    titolo: 'label livello',
    versioniAssegnate: [{
      id: 100,
      ritirata: false,
      in_corso: false,
      consegnata: false,
      titolo: 'titolo versione 1',
      assegnata_data: '11/03/2018',
      assegnazione: 1000,
      totale_domande: 12,
    }, {
      id: 200,
      ritirata: false,
      in_corso: true,
      consegnata: false,
      titolo: 'titolo versione 2',
      assegnata_data: '01/04/2018',
      assegnazione: 2000,
      totale_domande: 13,
    }, {
      id: 300,
      ritirata: false,
      in_corso: false,
      consegnata: true,
      titolo: 'titolo versione 3',
      assegnata_data: '11/05/2018',
      assegnazione: 3000,
      totale_domande: 14,
    }, {
      id: 400,
      ritirata: true,
      in_corso: true,
      consegnata: false,
      titolo: 'titolo versione 4',
      assegnata_data: '30/06/2018',
      assegnazione: 4000,
      totale_domande: 15,
    }, {
      id: 500,
      ritirata: true,
      in_corso: true,
      consegnata: true,
      titolo: 'titolo versione 5',
      assegnata_data: '21/07/2018',
      assegnazione: 5000,
      totale_domande: 16,
    }],
  },
  onVersioneLivelloFetch: () => { },
  onInitializeEsecuzione: () => { },
  dispatch: () => { },
  userAnagraphics: {
    id: 666,
  },
  userAppData: {
    hints: {
      versione_G: true,
      stepUP: true,
    },
  },
};


describe('<VersioniStudenteView />', () => {
  it('!spinner !feedback.hasFeedback', () => {
    const renderedComponent = shallow(
      <VersioniStudenteView {...mockProps} />
    );

    expect((renderedComponent).find(Spinner).length).toBe(0);
    expect((renderedComponent).find(AlertBanner).length).toBe(0);
    expect((renderedComponent).find(VersioniLivelloOverview).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('spinner !feedback.hasFeedback', () => {
    const props = {
      ...mockProps,
      spinner: true,
    };
    const renderedComponent = shallow(
      <VersioniStudenteView {...props} />
    );

    expect((renderedComponent).find(Spinner).length).toBe(1);
    expect((renderedComponent).find(AlertBanner).length).toBe(0);
    expect((renderedComponent).find(VersioniLivelloOverview).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('!spinner feedback.hasFeedback', () => {
    const props = {
      ...mockProps,
      feedback: {
        ...mockProps.feedback,
        hasFeedback: true,
      },
    };
    const renderedComponent = shallow(
      <VersioniStudenteView {...props} />
    );

    expect((renderedComponent).find(Spinner).length).toBe(0);
    expect((renderedComponent).find(AlertBanner).length).toBe(1);
    expect((renderedComponent).find(VersioniLivelloOverview).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('componentDidMount deve chiamare onVersioneLivelloFetch', () => {
    const props = {
      ...mockProps,
      onVersioneLivelloFetch: jest.fn(),
    };
    const spy = jest.spyOn(VersioniStudenteView.prototype, 'componentDidMount');
    const renderedComponent = shallow(
      <VersioniStudenteView {...props} />
    );

    renderedComponent.instance();
    expect(spy).toHaveBeenCalled();
    expect(props.onVersioneLivelloFetch).toHaveBeenCalledWith('123', false);
  });

  it('eseguiFunction deve chiamare onInitializeEsecuzione', () => {
    const props = {
      ...mockProps,
      onInitializeEsecuzione: jest.fn(),
    };
    const renderedComponent = shallow(
      <VersioniStudenteView {...props} />
    );
    const instance = renderedComponent.instance();
    expect(props.onInitializeEsecuzione).not.toHaveBeenCalled();

    instance.eseguiFunction({
      assegnazione: 555,
      in_corso: true,
    });
    expect(props.onInitializeEsecuzione).toHaveBeenCalledWith({
      backUrl: '/versioni-livello/123',
      id: 555,
      isDocente: false,
      userHints: { stepUP: true, versione_G: true },
      history: mockProps.history,
      dispatchFunction: mockProps.dispatch,
      productName: 'alatin',
      inCorso: true,
      userId: 666,
    });
  });

  it('visualizzaStatistiche deve fare un history.push', () => {
    const props = {
      ...mockProps,
      history: {
        push: jest.fn(),
      },
    };

    const renderedComponent = shallow(
      <VersioniStudenteView {...props} />
    );
    const instance = renderedComponent.instance();
    expect(props.history.push).not.toHaveBeenCalled();
    instance.visualizzaStatistiche();
    expect(props.history.push).toHaveBeenCalledWith('/versione-livello-statistiche/123');
  });
});


describe('<VersioneStudente />', () => {
  const withReducerConfiguration = injectReducer({ key: 'configuration', reducer: reducerCurryer(mockConfiguration) });
  const withReducerUser = injectReducer({ key: 'user', reducer: userReducer });
  const withReducerVersione = injectReducer({ key: 'versioni', reducer: versioneReducer });

  const ConfigurationWrapper = compose(
    withReducerConfiguration,
    withReducerUser,
    withReducerVersione,
    connect()
  )(VersioneStudente);

  it('le props iniziali devono essere settate correttamente', () => {
    const store = configureStore({}, {});
    const renderedComponent = shallow(
      <ConfigurationWrapper
        match={{ params: { id: '123' } }}
        history={{ push: () => { } }}
      />, {
        context: { store },
      }
    ).dive().dive().dive().dive().dive();

    const instance = renderedComponent.instance();
    const receivedProps = instance.props;

    expect(receivedProps.configuration).toEqual(mockConfiguration);
    expect(receivedProps.userAppData).toEqual(defaultAppData.toJS());
    expect(receivedProps.spinner).toBe(false);
    expect(receivedProps.feedback).toEqual(defaultVersioneFeedback.toJS());
    expect(receivedProps.versioniLivello).toEqual(defaultVersioniLivello.toJS());
  });

  it('testo il corretto funzionemento delle funzioni del mapDispatchToProps', () => {
    const mockStore = configureStore({}, {});
    mockStore.dispatch = jest.fn();

    const renderedComponent = shallow(
      <ConfigurationWrapper
        match={{ params: { id: '123' } }}
        history={{ push: () => { } }}
      />, {
        context: { store: mockStore },
      }
    ).dive().dive().dive().dive().dive();

    const instance = renderedComponent.instance();
    const receivedProps = instance.props;

    receivedProps.onVersioneLivelloFetch(1, true, 456);
    expect(mockStore.dispatch).toHaveBeenCalledWith(
      versioniLivelloFetch(1, true, 456)
    );

    receivedProps.onInitializeEsecuzione({ data: 123 });
    expect(mockStore.dispatch).toHaveBeenCalledWith(
      versioniVersioneEsecuzioneInitialize({ data: 123 })
    );
  });
});
