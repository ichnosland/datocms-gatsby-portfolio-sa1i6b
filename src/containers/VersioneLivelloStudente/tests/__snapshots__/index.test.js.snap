// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VersioniStudenteView /> !spinner !feedback.hasFeedback 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="label livello"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <VersioniLivelloStudenteOverview
      eseguiFunction={[Function]}
      versioniAssegnate={
        Array [
          Object {
            "assegnata_data": "11/03/2018",
            "assegnazione": 1000,
            "consegnata": false,
            "id": 100,
            "in_corso": false,
            "ritirata": false,
            "titolo": "titolo versione 1",
            "totale_domande": 12,
          },
          Object {
            "assegnata_data": "01/04/2018",
            "assegnazione": 2000,
            "consegnata": false,
            "id": 200,
            "in_corso": true,
            "ritirata": false,
            "titolo": "titolo versione 2",
            "totale_domande": 13,
          },
          Object {
            "assegnata_data": "11/05/2018",
            "assegnazione": 3000,
            "consegnata": true,
            "id": 300,
            "in_corso": false,
            "ritirata": false,
            "titolo": "titolo versione 3",
            "totale_domande": 14,
          },
          Object {
            "assegnata_data": "30/06/2018",
            "assegnazione": 4000,
            "consegnata": false,
            "id": 400,
            "in_corso": true,
            "ritirata": true,
            "titolo": "titolo versione 4",
            "totale_domande": 15,
          },
          Object {
            "assegnata_data": "21/07/2018",
            "assegnazione": 5000,
            "consegnata": true,
            "id": 500,
            "in_corso": true,
            "ritirata": true,
            "titolo": "titolo versione 5",
            "totale_domande": 16,
          },
        ]
      }
      visualizzaStatisticheFunction={[Function]}
    />
  </Container>
</Page>
`;

exports[`<VersioniStudenteView /> !spinner feedback.hasFeedback 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="label livello"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <AlertBanner
      actioncolor=""
      theme={
        Object {
          "brand": "#00BBEF",
          "radius": Object {
            "general": "6px",
          },
        }
      }
    />
  </Container>
</Page>
`;

exports[`<VersioniStudenteView /> spinner !feedback.hasFeedback 1`] = `
<Page>
  <TopBar
    backArrow={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    backNav={
      Object {
        "enabled": true,
        "url": "/homepage",
      }
    }
    closeBtn={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    lesson={
      Object {
        "enabled": false,
        "url": "",
      }
    }
    pinned={true}
    single={true}
    title="label livello"
    topSwitcher={
      Object {
        "items": Array [],
      }
    }
  />
  <Container>
    <Spinner
      number={3}
    />
  </Container>
</Page>
`;
